src/main.rs

```diff
@@ -204,12 +201,15 @@ fn find_available_region(mem_map: &HwMemMap, size: u64) -> Option<SupervisorPage
 // Creates the Sv48 page table based on the accessible regions of memory in the provided memory
 // map.
 fn setup_hyp_paging(hyp_map: HypMap, hyp_mem: &mut HypPageAlloc) {
-    let page_table = hyp_map.new_page_table(hyp_mem);
-    let satp = page_table.satp();
-    // Store the SATP value for other CPUs. They load from the global in start_secondary.
-    SATP_VAL.call_once(|| satp);
-    // Install the page table in satp
-    CSR.satp.set(satp);
+    // Create per-cpu page tables.
+    let cpu_info = CpuInfo::get();
+    for i in 0..cpu_info.num_cpus() {
+        let page_table = hyp_map.new_page_table(hyp_mem);
+        PerCpu::set_cpu_page_table(CpuId::new(i), page_table);
+    }
```

Comment:
Instead of setting up all the secondary CPU PTs on the boot CPU, could we stick `hyp_map` in a global and let each secondary CPU set up its own PT? That would (theoretically) allow us to set up PTs in parallel and avoids having to reach into the `PerCpu` struct from the boot CPU (obviously fine here, but a little ugly).

Reply:
I could. Could be ugly in other forms. `hyp_mem` must also be stored somewhere. `hyp_mem` is currently consumed by the host vm loader.
Now in the current code we wait for the host_vm to be finalized, so we could just stick the `hyp_mem` in a global, and pass it only as a mutable reference. And impose that no one uses it after. I think that letting the loader consume it is much better than this scenario.
Alternatively we could re-introduce `pte_page_count`, _over_-allocate the pages on BSP for all cpus in a sequential page and using that -- in a thread-safe way. Which means that `new_page_table` will ascept a SequentialPages.

I am open to what you think it's best, I just think that this current code is the slighlty better version.

Please also note that I would like to keep the cpu_page_table field to map per-cpu mapping in the future.

Reply:
> Now in the current code we wait for the host_vm to be finalized, so we could just stick the hyp_mem in a global, and pass it only as a mutable reference. And impose that no one uses it after. I think that letting the loader consume it is much better than this scenario.

Yeah ok, that's a fair point. What you have now SGTM then.